core/src/main/scala/org/apache/spark/rdd/AsyncRDDActions.scala

```diff
@@ -84,18 +87,18 @@ class AsyncRDDActions[T: ClassTag](self: RDD[T]) extends Serializable with Loggi
       } else {
         // The number of partitions to try in this iteration. It is ok for this number to be
         // greater than totalParts because we actually cap it at totalParts in runJob.
-        var numPartsToTry = 1L
+        var numPartsToTry = Math.max(self.conf.get(RDD_INITIAL_NUM_PARTITIONS), 1)
         if (partsScanned > 0) {
           // If we didn't find any rows after the previous iteration, quadruple and retry.
           // Otherwise, interpolate the number of partitions we need to try, but overestimate it
           // by 50%. We also cap the estimation in the end.
-          if (results.size == 0) {
-            numPartsToTry = partsScanned * 4L
+          if (results.isEmpty) {
+            numPartsToTry = partsScanned * scaleUpFactor
```

Comment:
It looks like this is fixing a pre-existing bug where the `RDD_LIMIT_SCALE_UP_FACTOR` wasn't being applied to the AsyncRDDActions version of `take()`. Nice!